lib/pages/twake_welcome/twake_welcome.dart

```diff
@@ -58,8 +61,10 @@ class TwakeWelcomeController extends State<TwakeWelcome> with ConnectPageMixin {
 
   MatrixState get matrix => Matrix.of(context);
 
-  void onClickSignIn() {
+  void onClickSignIn() async {
     Logs().d("TwakeIdController::onClickSignIn: Login Url - $loginUrl");
+    final homeserverExisted = await _homeserverExisted();
+    if (homeserverExisted) return;
     _redirectRegistrationUrl(loginUrl);
```

Comment:
Found it same with logic inside `onClickCreateTwakeId()`. How about reuse?

Reply:
Updated it